Forgot password

EricChavarria opened the pull request:
I made the forgot password page. When they submit it it will give them a confirmation message on the screen and when we get emailing to work we will send the password reset email at this time. Also I realized we need to make a password reset page as well
<img width="324" alt="Screen Shot 2020-02-16 at 12 50 30 PM" src="https://user-images.githubusercontent.com/27781451/74609786-6f061f00-50bb-11ea-8094-3c01943ad00d.png">
<img width="332" alt="Screen Shot 2020-02-16 at 12 52 09 PM" src="https://user-images.githubusercontent.com/27781451/74609790-74636980-50bb-11ea-90bf-57aeca73af05.png">

Commit:
Made the contact page

Commit:
Make conditional class

Commit:
Create Services Page

Commit:
get rid of id's

Haon1919 committed:
Merge branch 'develop' into contact

Commit:
Use the FormInput Component

Haon1919 committed:
Calendar html + render helper

Begin html and add a render function to construct the calendar with proper dates for the respective month.

joverton2697 committed:
Landing Page

Landing page is mostly finished, final images and text need to be decided. Header has blank space underneath

joverton2697 committed:
Implementing feedback

Commit:
Make changes to Contact Page

Commit:
Make styling change to services

Commit:
Change content

Commit:
Get started on Blog Page

Commit:
Finish up Blog Component

Commit:
get Started on Forgot Password Page

Commit:
Finish Forgot Password Page

Haon1919 commented:
You should have the button say reset password

Haon1919 commented on src/pages/ForgotPassword.tsx:
```diff
@@ -16,12 +22,44 @@ export const ForgotPassword = () => {
+                </FormInput>
+                <input className={isDisabled ? `${styles.send_forgot_password_form} ${styles.disabled}` : `${styles.send_forgot_password_form}`}  type="submit" value="Send Password Reset Link" disabled={isDisabled}/>
+            </form>
+            <h1 className={styles.submitted_forgot_password_form_text} style={{"display":(forgotPasswordFormSubmitted ? "block" : "none")}}>Your form has been submitted. If there is an account with the email you entered you should get an email to reset your password.</h1>
```
Conditionally render this instead of setting the style to display none like that. It looks like {forgotPasswordSubmitted && <h1>...}

Haon1919 committed:
Got the repository battle ready

Haon1919 committed:
Got the repository battle ready

Finish calendar, add schedule lesson and lesson page, fix header z-index issue, add fake login functionality.

Haon1919 committed:
Merge remote-tracking branch 'origin/Calendar' into Calendar

Haon1919 committed:
Merge remote-tracking branch 'origin/blog' into PresentationBranch

Haon1919 committed:
Merge branch 'forgot-password' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Services' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/contact' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Landing' into PresentationBranch

Haon1919 committed:
Fix Login link in header

Fix login link in header by prefacing it with a slash

Commit:
Fix typo and mini bug

Haon1919 committed:
General tab remains collapsed when user logs out

Fix bug where if the user logs out and the Students tab is open, the general tab remains collapsed.

Haon1919 committed:
Merge remote-tracking branch 'origin/PresentationBranch' into PresentationBranch

Haon1919 committed:
Nonrequired form blocking submit if fails validation

If a non required field fails validation, make sure that when they remove the text it removes the failing field from the list of failing fields.

Haon1919 committed:
Minor form failing error persists for non minors

Fix error where when the birth date of the user made them a minor,and the user entered an invalid email for the guardian email then switched the birth date to make them an adult and filled in all of the other forms correctly, the  invalid guardian email still disallowed them from submitting even though they guardian email is not an option for adults.

Haon1919 committed:
Merge branch 'PresentationBranch' into forgot-password
